elytron/src/test/java/org/wildfly/extension/elytron/SyslogAuditLogTestCase.java

```diff
@@ -0,0 +1,334 @@
+     */
+    @Test
+    public void testNumberedReconnectAttemptsGoodHost() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.RECONNECT_ATTEMPTS).set(RECONNECT_NUMBER);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the server successfully sends a message with reconnect-attempts set to -1 (infinite reconnects)
+     */
+    @Test
+    public void testInfiniteReconnectAttemptsGoodHost() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.RECONNECT_ATTEMPTS).set(-1);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+    /**
+     * Configures the required parameters for the syslog operation
+     */
+    private ModelNode createSyslogOperation(int port, String transport, String serverAddress) {
+        ModelNode operation = new ModelNode();
+        operation.get(ClientConstants.OP_ADDR).add("subsystem","elytron").add("syslog-audit-log", "syslog-test");
+        operation.get(ClientConstants.OP).set(ClientConstants.ADD);
+        operation.get(ElytronDescriptionConstants.SERVER_ADDRESS).set(serverAddress);
+        operation.get(ElytronDescriptionConstants.PORT).set(Integer.toString(port));
+        operation.get(ElytronDescriptionConstants.TRANSPORT).set(transport);
+        operation.get(ElytronDescriptionConstants.HOST_NAME).set(HOST_NAME);
+        return operation;
+    }
+
+    /**
+     * Calls the syslog operation creator with a given address using UDP
+     */
+    private ModelNode createUdpSyslogOperation(String serverAddress) {
+        return createSyslogOperation(UDP_PORT, UDP_TRANSPORT, serverAddress);
+    }
+
+    /**
+     * Asserts that the expected failure happened
+     *
+     * @param response The response from the operation
+     * @param messages The array of messages that should be contained in the response
+     */
+    private void assertCorrectError(ModelNode response, String[] messages) {
+        for (String msg : messages) {
+            assertTrue(response.get(FAILURE_DESCRIPTION).asString().contains(msg));
+        }
+    }
+
+    /**
+     * Verifies the operation was successful
+     */
+    private ModelNode assertSuccess(ModelNode response) {
+        if (!response.get(OUTCOME).asString().equals(SUCCESS)) {
+            Assert.fail(response.toJSONString(false));
+        }
+        return response;
+    }
+
+    /**
+     * Verifies the operation failed
+     */
+    private ModelNode assertFailed(ModelNode response) {
+        if (! response.get(OUTCOME).asString().equals(FAILED)) {
+            Assert.fail(response.toJSONString(false));
+        }
+        return response;
+    }
+
+    /**
+     * Verifies the operation cancelled before completion
+     */
+    private ModelNode assertCancelled(ModelNode response) {
```

Comment:
Isn't the method name misleading? I would assume that it throws some kind of exception based on its name.  Wouldn't also this approach mute any previous responses which are not cancelled and leave only the last one which would then be in `failString`? Does it make sense to continue in test if any of requests hasn't been cancelled? (I'm referencing https://github.com/wildfly/wildfly-core/pull/3856/files#diff-db84c0d398e11d3e59f5b01d0367d032R202)

Reply:
The method name is based on what the ModelNode response string contains ("cancelled"), with the naming being to keep it in line with the other two response checks (assertFailed and assertSuccess). Only one response would be created, since it is in the executeOperation that the child thread waits (while it attempts to send the initial connection message). If the response seen in assertCancelled is not "cancelled" (meaning that there was not infinite reconnect attempts since otherwise an exception would be thrown) then failString will be set, causing the test to fail due to failString being non empty after the threads are joined. Otherwise, failString will get set due to an exception being thrown during the threads execution which will also cause the test to fail after the threads are joined.

Reply:
Thank you. When I was talking about misleading name I had the "assert" part in mind.

Yes, `failString` is set, but if there was more than one error (result with status set to anything else than "cancelled"), we can see only last error.

Reply:
I'm not sure that I see the case where it gets set, due to the only cases that it gets set being (with only one of the following scenarios being possible in any given test run):
1. Once if there is an unexpected exception that gets thrown during the infinite attempts while executeOperation is running (where the infinite attempts happens inside of)
2. After the response gets set, which will occur only one time after the parent thread interrupts the child thread (the expected outcome of failString being kept empty due to assertCancelled seeing the status as cancelled and so skipping over setting the value) or if the message gets successfully sent (which should never happen regardless due to the invalid IP)/if the operation fails due to some other issue, in either case assertCancelled will be called once and will set the failString value with the bad outcome and then the child thread will end before the interruption
In either of the above cases, the child thread will no longer be executing once failString gets set. What's the situation you're thinking of where failString will get set multiple times?

Reply:
Ah, my bad. I was reading the code wrong. Thank you.